src/component/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx

```diff
@@ -94,28 +111,118 @@ export class TimeLayerSliderPanel extends React.Component<TimeLayerSliderPanelPr
     // binds
     this.onTimeChanged = this.onTimeChanged.bind(this);
     this.autoPlay = this.autoPlay.bind(this);
-    this.onDataRangeOk = this.onDataRangeOk.bind(this);
+    this.updateDataRange = this.updateDataRange.bind(this);
   }
 
-  componentDidUpdate() {
-    this.wrapTimeSlider();
+  componentDidUpdate(prevProps: TimeLayerSliderPanelProps) {
+    if (!(_isEqual(prevProps.timeAwareLayers, this.props.timeAwareLayers))) {
+      // update slider properties if some another layer set was chosen
+      this.wrapTimeSlider();
+      this.findRangeForLayers();
+    }
+  }
+
+  /**
+   *
+   * @param nextProps
+   * @param nextState
+   */
+  shouldComponentUpdate(nextProps: TimeLayerSliderPanelProps, nextState: TimeLayerSliderPanelState) {
+    const {
+      value,
+      autoPlayActive
+    } = this.state;
+    const {
+      startDate,
+      endDate,
+      timeAwareLayers
+    } = this.props;
+
+    if (nextState.value !== value) {
+      return true;
+    }
+    if (nextState.autoPlayActive !== autoPlayActive) {
+      return true;
+    }
+    if (nextProps.startDate !== startDate) {
+      return true;
+    }
+    if (nextProps.endDate !== endDate) {
+      return true;
+    }
+    if (!(_isEqual(nextProps.timeAwareLayers, timeAwareLayers))) {
```

Comment:
Is that a deep equal? If so, this might eat performance

Reply:
The current static mock configuration from some another client, which is currently used to test this component, offers maximal 6 time aware layers which can be passed here, so the performance is not really influenced.

But you're probably right, I've added a `TODO` for now to possibly recheck this later as soon as a real scenario will exist.